module/input/SensorFilter/src/MotionModel.hpp

```diff
@@ -150,23 +145,17 @@ namespace module::input {
             // UPDATE ANGULAR POSITION/VELOCITY
             // ********************************
 
-            // Extract our unit quaternion rotation
-            const Eigen::Quaternion<Scalar> Rwt(newState.Rwt.normalized());
-
             // Apply our rotational velocity to our orientation
-            // The change in the quaternion q is (1/2)*omega*q
-            // Here is an explanation https://gamedev.stackexchange.com/a/157018
-            // Here is another derivation https://fgiesen.wordpress.com/2012/08/24/quaternion-differentiation/
-            // dq/dt = (1/2)*omega*Rwt
-            const Eigen::Quaternion<Scalar> dq_dt = Eigen::Quaternion<Scalar>(0.0,
-                                                                              newState.omegaTTt.x() * 0.5,
-                                                                              newState.omegaTTt.y() * 0.5,
-                                                                              newState.omegaTTt.z() * 0.5)
-                                                    * Rwt;
-            // The change in the rotation is the derivative times the differential (which is the time-step)
-            const Eigen::Quaternion<Scalar> change = Eigen::Quaternion<Scalar>(deltaT * dq_dt.coeffs());
-            // We can add the change to the original, as long as our time step is small enough
-            newState.Rwt = Eigen::Quaternion<Scalar>(Rwt.coeffs() + change.coeffs()).normalized();
+            // If the norm of the gyroscope update is 0 then there is not update needed
+            const Scalar norm = newState.omegaTTt.norm();
+            if (norm != Scalar(0)) {
+                // The gyroscope has measured a rotation of norm * deltaT around the axis
+                // omegaTTt / norm
+                Eigen::AngleAxis dq(norm * deltaT, newState.omegaTTt / norm);
```

Comment:
This seems way too easy lmfaooo

Reply:
I know right. But someome much smarter than me told me it's correct

Reply:
that's good enough for me :^)